lib/Net/SAML2/IdP.pm

```diff
@@ -164,50 +165,39 @@ sub new_from_xml {
         }
     }
 
-    my @certs = ();
+    my %certs = ();
 
-    for my $key (
-        $xpath->findnodes('//md:EntityDescriptor/md:IDPSSODescriptor/md:KeyDescriptor'))
-    {
-        my @uses;
-        push (@uses, $key->getAttribute('use') || 'signing');
-        push (@uses, 'encryption') if !$key->getAttribute('use');
-
-
-        $key->setNamespace('http://www.w3.org/2000/09/xmldsig#', 'ds');
-
-        my ($text)
-            = $key->findvalue("ds:KeyInfo/ds:X509Data/ds:X509Certificate", $key)
-            =~ /^\s*(.+?)\s*$/s;
-
-        # rewrap the base64 data from the metadata; it may not
-        # be wrapped at 64 characters as PEM requires
-        $text =~ s/\n//g;
-
-        my @lines;
-        while(length $text > 64) {
-            push @lines, substr $text, 0, 64, '';
+    my @nodes = $xpath->findnodes('//md:EntityDescriptor/md:IDPSSODescriptor/md:KeyDescriptor');
+    if (@nodes == 1) {
+        my $key = $nodes[0];
+        my $uses = $key->getAttribute('use');
+        my $pem = $class->_get_pem_from_keynode($key);
+        if (!$uses) {
+            $certs{signing} = [$pem];
+            $certs{encryption} = [$pem];
         }
-        push @lines, $text;
-
-        $text = join "\n", @lines;
+        else {
+            $certs{$uses} = [$pem];
+        }
+    }
+    else {
+        for my $key (@nodes) {
+            my $uses = $key->getAttribute('use');
+            if (!$uses) {
```

Comment:
https://docs.oasis-open.org/security/saml/v2.0/saml-schema-metadata-2.0.xsd use is actually optional.

`<attribute name="use" type="md:KeyTypes" use="optional"/>`